src/harness/testcases/WINNF_FT_S_PPR_testcase.py

```diff
@@ -401,3 +409,317 @@ def test_WINNF_FT_S_PPR_2(self, config_filename):
+    ppa_record_1, pal_records_1 = makePpaAndPalRecordsConsistent(
+        ppa_record_1, [pal_record_1], pal_low_frequency, pal_high_frequency,
+        'test_user_2')
+
+    # Generate FAD records.
+    cbsd_records = [device_a, device_c, device_e]
+    # Create CBSD reference IDs.
+    cbsd_reference_id_a = generateCbsdReferenceId(device_a['fccId'],
+                                                  device_a['cbsdSerialNumber'])
+    cbsd_reference_id_c = generateCbsdReferenceId(device_c['fccId'],
+                                                  device_c['cbsdSerialNumber'])
+    cbsd_reference_id_e = generateCbsdReferenceId(device_e['fccId'],
+                                                  device_e['cbsdSerialNumber'])
+    cbsd_reference_ids = [[
+        cbsd_reference_id_a, cbsd_reference_id_c, cbsd_reference_id_e
+    ]]
+    grant_record_list = [[grant_a], [grant_c], [grant_e]]
+    cbsd_records = generateCbsdRecords(cbsd_records, grant_record_list)
+    for cbsd in cbsd_records:
+      for grant in cbsd['grants']:
+        grant['channelType'] = 'PAL'
+    # Create records.
+    sas_harness_dump_records = {
+        'cbsdRecords': cbsd_records,
+        'ppaRecords': generatePpaRecords([ppa_record_1], cbsd_reference_ids),
+    }
+
+    # SAS test harness configuration.
+    sas_harness_config = {
+        'sasTestHarnessName': 'SAS-TH-1',
+        'hostName': getFqdnLocalhost(),
+        'port': getUnusedPort(),
+        'serverCert': getCertFilename('sas.cert'),
+        'serverKey': getCertFilename('sas.key'),
+        'caCert': 'certs/ca.cert'
+    }
+
+    config = {
+        'domainProxy':
+            domain_proxy,  # Includes registration and grant requests.
+        'ppaRecord': ppa_record_0,  # PPA in SAS UUT.
+        'ppaClusterList':
+            ppa_cluster_list,  # Same format and semantics as SIQ.12.
+        'palRecords': [pal_records_0[0],
+                       pal_records_1[0]],  # PALs for both PPAs.
+        'sasTestHarnessDumpRecords':
+            sas_harness_dump_records,  # CBSDs and one PPA.
+        'sasTestHarnessConfig':
+            sas_harness_config,  # Just the config, no records.
+    }
+    writeConfig(filename, config)
+
+  @configurable_testcase(generate_PPR_3_default_config)
+  def test_WINNF_FT_S_PPR_3(self, config_filename):
+    config = loadConfig(config_filename)
+
+    # Light config checking.
+    self.assertValidConfig(
+        config, {
+            'domainProxy': dict,
+            'ppaRecord': dict,
+            'ppaClusterList': list,
+            'palRecords': list,
+            'sasTestHarnessDumpRecords': dict,
+            'sasTestHarnessConfig': dict
+        })
+    self.assertEqual(
+        len(config['sasTestHarnessDumpRecords']['ppaRecords']), 1,
+        'Only one PPA is supported.')
+    # Make sure ID formats are correct.
+    ppa = config['sasTestHarnessDumpRecords']['ppaRecords'][0]
+    for cbsd_ref_id in ppa['ppaInfo']['cbsdReferenceId']:
```

Comment:
Do you think we need an assertion for Length as well for `ppa['ppaInfo']['cbsdReferenceId']`?

Reply:
Done